pkg/api/mlflow/service/metric/service.go

```diff
@@ -19,102 +19,62 @@ import (
 )
 
 func GetMetricHistory(c *fiber.Ctx) error {
-	id := c.Query("run_id", c.Query("run_uuid"))
-	key := c.Query("metric_key")
+	req := request.GetMetricHistoryRequest{}
+	if err := c.QueryParser(&req); err != nil {
+		return api.NewBadRequestError(err.Error())
+	}
 
-	log.Debugf("GetMetricHistory request: run_id=%q, metric_key=%q", id, key)
+	log.Debugf("GetMetricHistory request: run_id=%q, run_uuid=%q metric_key=%q", req.RunID, req.RunUUID, req.MetricKey)
 
-	if id == "" {
-		return api.NewInvalidParameterValueError("Missing value for required parameter 'run_id'")
-	}
-	if key == "" {
-		return api.NewInvalidParameterValueError("Missing value for required parameter 'metric_key'")
+	if err := ValidateGetMetricHistoryRequest(&req); err != nil {
+		return err
 	}
 
 	var metrics []database.Metric
-	if tx := database.DB.Where("run_uuid = ?", id).Where("key = ?", key).Find(&metrics); tx.Error != nil {
-		return api.NewInternalError("Unable to get metric history for metric %q of run %q", key, id)
+	if err := database.DB.Where(
+		"run_uuid = ?", req.GetRunID(),
+	).Where(
+		"key = ?", req.MetricKey,
+	).Find(&metrics).Error; err != nil {
+		return api.NewInternalError(
+			"Unable to get metric history for metric %q of run %q", req.MetricKey, req.GetRunID(),
+		)
 	}
 
-	resp := &response.GetMetricHistoryResponse{
-		Metrics: make([]response.MetricPartialResponse, len(metrics)),
-	}
-	for n, m := range metrics {
-		resp.Metrics[n] = response.MetricPartialResponse{
-			Key:       m.Key,
-			Value:     m.Value,
-			Timestamp: m.Timestamp,
-			Step:      m.Step,
-		}
-		if m.IsNan {
-			resp.Metrics[n].Value = "NaN"
-		}
-	}
+	resp := response.NewMetricHistoryResponse(metrics)
 
 	log.Debugf("GetMetricHistory response: %#v", resp)
 
 	return c.JSON(resp)
 }
 
 func GetMetricHistoryBulk(c *fiber.Ctx) error {
-	q := struct {
-		RunIDs     []string `query:"run_id"`
-		MetricKey  string   `query:"metric_key"`
-		MaxResults int      `query:"max_results"`
-	}{}
-
-	if err := c.QueryParser(&q); err != nil {
+	req := request.GetMetricHistoryBulkRequest{}
+	if err := c.QueryParser(&req); err != nil {
 		return api.NewBadRequestError(err.Error())
 	}
 
-	if len(q.RunIDs) == 0 {
-		return api.NewInvalidParameterValueError("GetMetricHistoryBulk request must specify at least one run_id.")
+	if err := ValidateGetMetricHistoryBulkRequest(&req); err != nil {
+		return err
 	}
+	log.Debugf("GetMetricHistoryBulk request: %#v", req)
```

Comment:
In some places we log the request before validation, and in others we log it after (this was the case _before_ your refactoring, my bad!). We should pick one and stick with it. Maybe logging before validation, as this is for debug purposes?

Reply:
yep, lets log everything before validation. I didn't check this moment yet but yep, nice catch so we can handle it right now. Let me change everything.